modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/GridCacheQueryDetailMetricsAdapter.java

```diff
@@ -157,6 +157,24 @@ public GridCacheQueryDetailMetricsAdapter aggregate(QueryDetailMetrics m) {
         );
     }
 
+    /**
+     * Aggregate metrics.
+     *
+     * @param m Other metrics to take into account.
+     * @return Aggregated metrics.
+     */
+    public GridCacheQueryDetailMetricsAdapter merge(QueryDetailMetrics m) {
```

Comment:
Changes in the base class it's a bad idea. Let's map to QueryDetailMetrics and move merge to it.

Reply:
I've moved merge to QueryDetailMetrics.